config/config_test.go

```diff
@@ -33,11 +33,28 @@ func TestDefaults(t *testing.T) {
 	}
 }
 
+func TestMarshalStaticContentRootDirectory(t *testing.T) {
+	testConf := Config{
+		Server: Server{
+			StaticContentRootDirectory: "/tmp",
+		},
+	}
+
+	yamlString, err := Marshal(&testConf)
+	if err != nil {
+		t.Errorf("Failed to marshal: %v", err)
+	}
+	if yamlString != "server:\n  static_content_root_directory: /tmp\n" {
+		t.Errorf("Failed to marshal - StaticContentRootDirectory to static_content_root_directory: [%v]", yamlString)
+	}
+}
```

Comment:
Looks good. This tests marshalling with the underscores. Now just add an unmarshal to make sure the underscores in the yaml goes the other way. Something like:

```
conf2, err := Unmarshal(yamlString)
if err != nil {
    t.Errorf("Failed to unmarshal: %v", err)
}
if (conf2.Server.StaticContentRootDirectory != "/tmp") {
    t.Errorf("Failed to unmarshal root dir")
}
```

I think the other test might confirm this, but this makes it more explicit what the test is trying to do. I specifically want to see the yaml with underscores getting marshaled then unmarshaled.

Reply:
> I think the other test might confirm this, but this makes it more explicit what the test is trying to do. I specifically want to see the yaml with underscores getting marshaled then unmarshaled.

Thought the same. Will make it explicit in the same test.